frame/system/src/lib.rs

```diff
@@ -711,13 +418,357 @@ decl_module! {
+	#[pallet::storage]
+	#[pallet::getter(fn extrinsic_data)]
+	pub(super) type ExtrinsicData<T: Config> =
+	StorageMap<_, Twox64Concat, u32, Vec<u8>, ValueQuery>;
+
+	/// The current block number being processed. Set by `execute_block`.
+	#[pallet::storage]
+	#[pallet::getter(fn block_number)]
+	pub(super) type Number<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+	/// Hash of the previous block.
+	#[pallet::storage]
+	#[pallet::getter(fn parent_hash)]
+	pub(super) type ParentHash<T: Config> = StorageValue<_, T::Hash, ValueQuery>;
+
+	/// Digest of the current block, also part of the block header.
+	#[pallet::storage]
+	#[pallet::getter(fn digest)]
+	pub(super) type Digest<T: Config> = StorageValue<_, DigestOf<T>, ValueQuery>;
+
+	/// Events deposited for the current block.
+	#[pallet::storage]
+	#[pallet::getter(fn events)]
+	pub(super) type Events<T: Config> =
+	StorageValue<_, Vec<EventRecord<T::Event, T::Hash>>, ValueQuery>;
+
+	/// The number of events in the `Events<T>` list.
+	#[pallet::storage]
+	#[pallet::getter(fn event_count)]
+	pub(super) type EventCount<T: Config> = StorageValue<_, EventIndex, ValueQuery>;
+
+	/// Mapping between a topic (represented by T::Hash) and a vector of indexes
+	/// of events in the `<Events<T>>` list.
+	///
+	/// All topic vectors have deterministic storage locations depending on the topic. This
+	/// allows light-clients to leverage the changes trie storage tracking mechanism and
+	/// in case of changes fetch the list of events of interest.
+	///
+	/// The value has the type `(T::BlockNumber, EventIndex)` because if we used only just
+	/// the `EventIndex` then in case if the topic has the same contents on the next block
+	/// no notification will be triggered thus the event might be lost.
+	#[pallet::storage]
+	#[pallet::getter(fn event_topics)]
+	pub(super) type EventTopics<T: Config> =
+	StorageMap<_, Blake2_128Concat, T::Hash, Vec<(T::BlockNumber, EventIndex)>, ValueQuery>;
+
+	/// Stores the `spec_version` and `spec_name` of when the last runtime upgrade happened.
+	#[pallet::storage]
+	pub type LastRuntimeUpgrade<T: Config> = StorageValue<_, LastRuntimeUpgradeInfo>;
+
+	/// True if we have upgraded so that `type RefCount` is `u32`. False (default) if not.
+	#[pallet::storage]
+	pub(super) type UpgradedToU32RefCount<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+	/// The execution phase of the block.
+	#[pallet::storage]
+	pub(super) type ExecutionPhase<T: Config> = StorageValue<_, Phase>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig {
+		pub changes_trie_config: Option<ChangesTrieConfiguration>,
+		#[serde(with = "sp_core::bytes")]
+		pub code: Vec<u8>,
+	}
+
+	#[cfg(feature = "std")]
+	impl Default for GenesisConfig {
+		fn default() -> Self {
+			Self {
+				changes_trie_config: Default::default(),
+				code: Default::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
```

Comment:
This whole method was generated by `PRINT_PALLET_UPGRADE=1 cargo check -p frame-system ` so assume it is correct. Though I do note it differs from the one I was using originally copied from the old branch:

https://github.com/paritytech/substrate/blob/gui-macro-attribute-everywhere/frame/system/src/lib.rs#L677-L690

Reply:
the template generate very ugly codes, because it just copy paste the generate code by decl_storage, but you should be able to refactor a lot.

Each first block are the individual builders of storages, and then there is the extra_genesis_builder.
I quickly checked it and it seems equivalent.

Reply:
Cool I will refactor this tomorrow!

Reply:
Refactored, and in the end it is the same as  the one I linked to on your branch.